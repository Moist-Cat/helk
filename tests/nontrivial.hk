let a = 5;
let b = 50;

let x = 10;

while (min(max(b - a, 0), 1)) {
    let x = x - 1;
    let b = b - a;
};

prints("We made this many iterations:");
print(10 - x);
prints("Final value of b:");
print(b);
