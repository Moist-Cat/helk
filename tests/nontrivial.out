We made this many iterations:
9.000000
Final value of b:
5.000000